cwms-data-api/src/main/java/cwms/cda/data/dao/watersupply/WaterSupplyUtils.java

```diff
@@ -158,4 +167,173 @@ public static WATER_USER_CONTRACT_TAB_T toWaterUserContractTs(WaterUserContract
+        List<WAT_USR_CONTRACT_ACCT_OBJ_T> watUsrContractAcctObjTList = new ArrayList<>();
+        for (Map.Entry<Instant, List<PumpTransfer>> entry : accounting.getPumpAccounting().entrySet()) {
+            for (PumpTransfer transfer : entry.getValue()) {
+                WAT_USR_CONTRACT_ACCT_OBJ_T watUsrContractAcctObjT = new WAT_USR_CONTRACT_ACCT_OBJ_T();
+                WATER_USER_CONTRACT_REF_T contractRef = toContractRef(accounting.getWaterUser(),
+                        accounting.getContractName());
+                watUsrContractAcctObjT.setWATER_USER_CONTRACT_REF(contractRef);
+                watUsrContractAcctObjT.setACCOUNTING_REMARKS(transfer.getComment());
+                watUsrContractAcctObjT.setPUMP_FLOW(transfer.getFlow());
+                LOOKUP_TYPE_OBJ_T transferType = toLookupTypeO(new LookupType.Builder()
+                        .withDisplayValue(transfer.getTransferTypeDisplay())
+                        .withActive(true)
+                        .withOfficeId(accounting.getWaterUser().getProjectId().getOfficeId())
+                        .build());
+                watUsrContractAcctObjT.setPHYSICAL_TRANSFER_TYPE(transferType);
+                switch (transfer.getPumpType()) {
+                    case IN:
+                        watUsrContractAcctObjT.setPUMP_LOCATION_REF(LocationUtil
+                                .getLocationRef(accounting.getPumpLocations().getPumpIn()));
+                        break;
+                    case OUT:
+                        watUsrContractAcctObjT.setPUMP_LOCATION_REF(LocationUtil
+                                .getLocationRef(accounting.getPumpLocations().getPumpOut()));
+                        break;
+                    case BELOW:
+                        watUsrContractAcctObjT.setPUMP_LOCATION_REF(LocationUtil
+                                .getLocationRef(accounting.getPumpLocations().getPumpBelow()));
+                        break;
+                    default:
+                        LOGGER.log(Level.WARNING, "Invalid pump type");
+                        break;
+                }
+                watUsrContractAcctObjT.setTRANSFER_START_DATETIME(Timestamp.from(entry.getKey()));
+                watUsrContractAcctObjTList.add(watUsrContractAcctObjT);
+            }
+        }
+        return new WAT_USR_CONTRACT_ACCT_TAB_T(watUsrContractAcctObjTList);
+    }
+
+    public static LOC_REF_TIME_WINDOW_TAB_T toTimeWindowTabT(WaterSupplyAccounting accounting) {
+        List<LOC_REF_TIME_WINDOW_OBJ_T> timeWindowList = new ArrayList<>();
+        for (Map.Entry<Instant, List<PumpTransfer>> entry : accounting.getPumpAccounting().entrySet()) {
+            for (PumpTransfer transfer : entry.getValue()) {
+                LOC_REF_TIME_WINDOW_OBJ_T timeWindow = new LOC_REF_TIME_WINDOW_OBJ_T();
+                switch (transfer.getPumpType()) {
+                    case IN:
+                        timeWindow.setLOCATION_REF(LocationUtil
```

Comment:
Same optimization recommendation here: move the location mapping up to the top so it doesn't need to be repeated

Reply:
Moved mapping to top of method